packages/block-library/src/site-logo/index.php

```diff
@@ -111,54 +111,52 @@ function _override_custom_logo_theme_mod( $custom_logo ) {
+function _sync_custom_logo_to_site_logo( $value ) {
+	if ( empty( $value ) ) {
 		delete_option( 'site_logo' );
 	} else {
-		remove_action( 'update_option_site_logo', '_sync_site_logo_to_custom_logo' );
-		update_option( 'site_logo', $value['custom_logo'] );
-		add_action( 'update_option_site_logo', '_sync_site_logo_to_custom_logo', 10, 2 );
+		update_option( 'site_logo', $value );
 	}
+
+	return $value;
 }
 
+add_filter( 'pre_set_theme_mod_custom_logo', '_sync_custom_logo_to_site_logo' );
+
 /**
- * Hooks `_sync_custom_logo_to_site_logo` in `update_option_theme_mods_$theme`.
+ * Deletes the site_logo when the custom_logo theme mod is removed.
  *
- * Runs on `setup_theme` to account for dynamically-switched themes in the Customizer.
+ * @param array $old_value Previous theme mod settings.
+ * @param array $value     Updated theme mod settings.
  */
-function _sync_custom_logo_to_site_logo_on_setup_theme() {
-	$theme = get_option( 'stylesheet' );
-	add_action( "update_option_theme_mods_$theme", '_sync_custom_logo_to_site_logo', 10, 2 );
+function _delete_site_logo_on_remove_custom_logo( $old_value, $value ) {
+	// If the custom_logo is being unset, it's being removed from theme mods.
+	if ( isset( $old_value['custom_logo'] ) && ! isset( $value['custom_logo'] ) ) {
+		delete_option( 'site_logo' );
+	}
 }
-add_action( 'setup_theme', '_sync_custom_logo_to_site_logo_on_setup_theme', 11 );
 
 /**
- * Updates the custom_logo theme-mod when the site_logo option gets updated.
- *
- * @param mixed $old_value The old option value.
- * @param mixed $value     The new option value.
- *
- * @return void
+ * Deletes the site logo when all theme mods are being removed.
  */
-function _sync_site_logo_to_custom_logo( $old_value, $value ) {
-	// Delete the option when the custom logo does not exist or was removed.
-	// This step ensures the option stays in sync.
-	if ( empty( $value ) ) {
-		remove_theme_mod( 'custom_logo' );
-	} else {
-		remove_filter( 'pre_set_theme_mod_custom_logo', '_sync_custom_logo_to_site_logo' );
-		set_theme_mod( 'custom_logo', $value );
-		add_filter( 'pre_set_theme_mod_custom_logo', '_sync_custom_logo_to_site_logo' );
+function _delete_site_logo_on_remove_theme_mods() {
+	if ( false !== get_theme_support( 'custom-logo' ) ) {
+		delete_option( 'site_logo' );
 	}
 }
 
-add_action( 'update_option_site_logo', '_sync_site_logo_to_custom_logo', 10, 2 );
+/**
+ * Hooks `_delete_site_logo_on_remove_custom_logo` in `update_option_theme_mods_$theme`.
+ * Hooks `_delete_site_logo_on_remove_theme_mods` in `delete_option_theme_mods_$theme`.
+ *
+ * Runs on `setup_theme` to account for dynamically-switched themes in the Customizer.
+ */
+function _delete_site_logo_on_remove_custom_logo_on_setup_theme() {
+	$theme = get_option( 'stylesheet' );
+	add_action( "update_option_theme_mods_$theme", '_delete_site_logo_on_remove_custom_logo', 10, 2 );
+	add_action( "delete_option_theme_mods_$theme", '_delete_site_logo_on_remove_theme_mods', 10, 0 );
```

Comment:
Can be simplified by removing the priority & args-number params:
```suggestion
	add_action( "delete_option_theme_mods_$theme", '_delete_site_logo_on_remove_theme_mods' );
```

Reply:
Thanks, I've updated this.